internal/provider/framework/provider.go

```diff
@@ -369,9 +369,13 @@ func (p *azureRmFrameworkProvider) Schema(_ context.Context, _ provider.SchemaRe
 									"delete_os_disk_on_deletion": schema.BoolAttribute{
 										Optional: true,
 									},
+
+									// TODO: Remove in 5.0
 									"graceful_shutdown": schema.BoolAttribute{
-										Optional: true,
+										Optional:           true,
+										DeprecationMessage: "'graceful_shutdown' has been deprecated and will be removed from v5.0 of the AzureRM provider.",
 									},
+
```

Comment:
This can be removed here and conditionally added based on the `FivePointOh()` flag at the bottom of this function:
```
	if !features.FivePointOh() {
		response.Schema.Blocks["features"].(schema.ListNestedBlock).NestedObject.Blocks["virtual_machine"].(schema.ListNestedBlock).NestedObject.Attributes["graceful_shutdown"] = schema.BoolAttribute{
			Optional:           true,
			DeprecationMessage: "'graceful_shutdown' has been deprecated and will be removed from v5.0 of the AzureRM provider.",
		}
	}

```
Sorry, can't add as a suggestion to use, due to deleted lines and requiring adding the `import` for the `features` package.

Reply:
Fixed.